### PR TITLE
feat(file_processors): add remote::unstructured-api provider

### PR DESCRIPTION
**Pull request by sahana-sreeram**
Add Unstructured.io as a remote file processor provider supporting 65+ file formats. Implementation follows the same pattern as remote::docling-serve provider with unit tests providing coverage

# What does this PR do?
Adds Unstructured.io as a remote file processor provider (`remote::unstructured-api`) supporting 65+ file formats including PDF, DOCX, PPTX, XLSX, EML, MSG, HTML, and more.

**Key capabilities:**
  - 65+ file format support
  - Email format support (EML/MSG) and DOC format support
  - Cloud-based processing via Unstructured.io SaaS API
  - Element type preservation (Title, NarrativeText, ListItem, Table, Image, etc.)
  - SOC2/HIPAA/GDPR certified processing

**Implementation:**
Follows the same architectural pattern as `remote::docling-serve` (PR #5412):
- Remote API integration using `unstructured-client` library
- Element-to-chunk mapping preserving semantic structure
- API key authentication via exporting `UNSTRUCTURED_API_KEY`
- Unit tests with mocked API responses (13 tests, same coverage pattern as docling-serve)

### 1. Unit Tests

Tests code logic in isolation by mocking API responses. Verifies the provider correctly validates inputs, calls the Unstructured API with proper authentication, maps elements to OGX chunks with all required metadata, and handles both direct file upload and file_id retrieval paths.

**Run:**
```bash
uv run pytest tests/unit/providers/file_processor/test_unstructured_api.py -v

Output:
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_rejects_no_file_and_no_file_id PASSED [  7%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_rejects_both_file_and_file_id PASSED [ 15%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_process_file_success PASSED [ 23%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_element_types_preserved PASSED [ 30%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_empty_elements_skipped PASSED [ 38%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_chunk_metadata_fields PASSED [ 46%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_chunk_id_uniqueness PASSED [ 53%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_page_numbers_preserved PASSED [ 61%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_token_count_calculated PASSED [ 69%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_process_file_via_file_id PASSED [ 76%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessor::test_api_key_used PASSED [ 84%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessorConfig::test_default_values PASSED [ 92%]
tests/unit/providers/file_processor/test_unstructured_api.py::TestUnstructuredApiFileProcessorConfig::test_sample_run_config PASSED [100%]

============================== 13 passed in 0.23s ==============================

Coverage:
- Input validation (2 tests)
- Core processing with element type preservation (3 tests)
- Chunk metadata mapping (4 tests)
- File ID retrieval path (1 test)
- API authentication (1 test)
- Configuration validation (2 tests)
```
### 2. Integration Test
Using a sample PDF, verifies the full end-to-end user flow from OGX server through the FileProcessor API to the real Unstructured.io API via exporting API key, confirming successful processing of text and PDF files with the correct chunk/embed structure, element type preservation, page number tracking, and error handling for invalid inputs.

```
Setup:
export UNSTRUCTURED_API_KEY="your-key"
uv run ogx stack run \
  --providers "file_processors=remote::unstructured-api,files=inline::localfs" \
  --port 8321

Test with curl:
curl -X POST http://localhost:8321/v1alpha/file-processors/process \
  -F "file=@syllabus_soc24_spring2018.pdf" | jq '.metadata, .chunks[0]'

Output:
{
  "processor": "unstructured-api",
  "processing_time_ms": 9872,
  "extraction_method": "unstructured-api",
  "file_size_bytes": 165750,
  "total_elements": 186
}
{
  "content": "Department of Sociology Harvard University Spring 2018",
  "chunk_id": "23d87af4-e79c-cfbf-1912-67c333904304",
  "metadata": {
    "document_id": "9410f8ab-459f-4459-9294-a4fc2f88ec8b",
    "element_type": "Title",
    "element_index": 0,
    "filename": "syllabus_soc24_spring2018.pdf",
    "page_number": 1
  },
  "chunk_metadata": {
    "chunk_id": "23d87af4-e79c-cfbf-1912-67c333904304",
    "document_id": "9410f8ab-459f-4459-9294-a4fc2f88ec8b",
    "source": "syllabus_soc24_spring2018.pdf",
    "created_timestamp": null,
    "updated_timestamp": null,
    "chunk_window": null,
    "chunk_tokenizer": null,
    "content_token_count": 7,
    "metadata_token_count": null
  }
}

Results:
- CHECK: Processed 11-page PDF → 186 chunks in 8.6 seconds
- CHECK: Element types detected: NarrativeText (104), Title (37), UncategorizedText (18), ListItem (17), Footer (10)
- CHECK: Page numbers preserved (1-11)
- CHECK: All required OGX Chunk fields populated correctly
```

Verification: existing pytests still pass

### CODE REVIEW

**Review by leseb — COMMENTED** [review]
please try to wire it with the "auto" provider

**Comment by alinaryan** [comment]
> please try to wire it with the "auto" provider

@leseb Is the intent to have auto choose between all available file processor backends? If so, I'm wondering why inline::docling and remote::docling-serve were not also wired in.

If the goal is to make auto a general dispatcher across providers, I'd prefer to do that in a followup PR that includes both docling and unstructured, so we can design the fallback mechanism properly rather than hard-coding one provider's config into auto's config class. For this PR, I think landing just the standalone remote::unstructured-api provider (without the auto changes) would be cleaner.

**Comment by leseb** [comment]
> > please try to wire it with the "auto" provider
> 
> @leseb Is the intent to have auto choose between all available file processor backends? If so, I'm wondering why inline::docling and remote::docling-serve were not also wired in.
> 
> If the goal is to make auto a general dispatcher across providers, I'd prefer to do that in a followup PR that includes both docling and unstructured, so we can design the fallback mechanism properly rather than hard-coding one provider's config into auto's config class. For this PR, I think landing just the standalone remote::unstructured-api provider (without the auto changes) would be cleaner.

🤔 they are wired, the auto provider just reads the format and forwards it to the appropriate provider. unless something changed recently :) 

i'm fine doing the wiring in a followup PR though!

**Comment by alinaryan** [comment]
> 🤔 they are wired, the auto provider just reads the format and forwards it to the appropriate provider. unless something changed recently :)

>i'm fine doing the wiring in a followup PR though!

@leseb I just checked, and docling isn't wired into auto today. Auto only imports and dispatches to pypdf and markitdown. Docling exists as standalone inline::docling and remote::docling-serve providers. Starter distribution only configures inline::auto for file_processors and auto has no docling references.

Sounds like we can do the wiring in a follow-up though, I can open an issue for this!

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    """
+
+    def __init__(self, config: UnstructuredApiFileProcessorConfig, files_api: Files) -> None:
+        self.config = config
+        self.files_api = files_api
+
+    async def process_file(
+        self,
+        request: ProcessFileRequest,
+        file: UploadFile | None = None,
```
Comment (cdoern) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py:46]: why do we allow file to be None?

File: src/ogx/providers/registry/file_processors.py [hunk 2]
```diff
@@ -191,6 +191,76 @@ def available_providers() -> list[ProviderSpec]:
 ## Documentation
 
 See [Docling Serve's documentation](https://github.com/docling-project/docling-serve/blob/main/docs/README.md) for more details on setup and configuration.
+""",
+        ),
+        RemoteProviderSpec(
+            api=Api.file_processors,
+            provider_type="remote::unstructured-api",
+            adapter_type="unstructured-api",
+            pip_packages=["unstructured-client>=0.44.0"],
```
Comment (cdoern) [inline src/ogx/providers/registry/file_processors.py]: this pip package is different from the one in the pyproject which has: `"unstructured-client>=0.25.0,<0.44.0"`

**Review by cdoern — COMMENTED** [review]
lgtm overall, two things I noticed!

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 1 above)
Comment (sahana-sreeram) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py:46]: I followed the `process_file()` from [remote::docling-serve](https://github.com/ogx-ai/ogx/blob/main/src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py), but also checked on this! The API supports two input modes
1. Direct upload: user provides file, so file_id is None
2. File reference: user provides file_id, so file is None

Option 2 would let users upload a file once through the endpoint /v1/files, then process multiple times (maybe with different providers or chunking strategies, i.e.) without re-uploading & just references the file_id. Lines 47-50 validate that only one of these processing options is provided

File: src/ogx/providers/registry/file_processors.py (hunk 2 above)
Comment (sahana-sreeram) [reply src/ogx/providers/registry/file_processors.py]: Just fixed in [1666a79](https://github.com/ogx-ai/ogx/pull/6076/commits/1666a79267bf6c5d21336a7a703b7235019dc9f8). Changed pip package registry to >=0.25.0,<0.44.0 to match pyproject.toml. unstructured-client>=0.44.0 requires pydantic>=2.12.5 but OGX uses <2.12.0.

File: src/ogx/providers/registry/file_processors.py [hunk 3]
```diff
@@ -191,6 +191,76 @@ def available_providers() -> list[ProviderSpec]:
 ## Documentation
 
 See [Docling Serve's documentation](https://github.com/docling-project/docling-serve/blob/main/docs/README.md) for more details on setup and configuration.
+""",
+        ),
+        RemoteProviderSpec(
+            api=Api.file_processors,
+            provider_type="remote::unstructured-api",
+            adapter_type="unstructured-api",
+            pip_packages=["unstructured-client>=0.25.0,<0.44.0"],
```
Comment (mattf) [inline src/ogx/providers/registry/file_processors.py]: nit - add a comment for why >= and why <.

was there a CVE in 0.24? is there an api break in 0.45?

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        # Call Unstructured.io API
+        try:
+            elements = await self._partition_via_api(content, filename)
+        except Exception as e:
+            log.error(
+                "Failed to process file with Unstructured.io API",
+                filename=filename,
+                error=str(e),
+            )
+            raise RuntimeError(f"Failed to process file with Unstructured.io API: {e}") from e
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: this error goes back to the user. please distinguish between errors the user can resolve, i.e. 40x errors like auth or a bad param, vs ones that are internal service errors.

**Review by mattf — DISMISSED** [review]

File: src/ogx/providers/registry/file_processors.py (hunk 3 above)
Comment (sahana-sreeram) [reply src/ogx/providers/registry/file_processors.py]: Added a comment line explaining version constraint:
  >=0.25.0+ supports full feature set (chunking + split_pdf_page_range) for possible future config expansion, which didn't exist in prev versions
  <0.44.0: pydantic compatibility (0.44.0+ requires pydantic >=2.12.5, OGX uses <2.12.0)

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 4 above)
Comment (sahana-sreeram) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: I decided to just use the SDK, which I found actually handles these errors and raises the right error number with description. Removed my manual error wrapping, which was generic. This also matches the error handling pattern in docling_serve.py (lines 125, 182).

**Manual testing output via OGX stack:**
  
  Started the stack with unstructured processor and a fake API key:
  
  ```bash
  curl -X POST http://localhost:8321/v1alpha/file-processors/process \
    -F "file_id=file-2c60593f183449a9a31d32299a192477" \
    -v 
    
  Response shows the error is correctly surfaced:

  < HTTP/1.1 401 Unauthorized
  < date: Mon, 15 Jun 2026 19:12:48 GMT
  < server: uvicorn
  < content-length: 149
  < content-type: application/json
  
  {"error":{"message":"API error occurred: Status 401. Body: {\"detail\":\"API key is malformed, please type the API key
  correctly in the header.\"}"}}
```
  Users can see:
  - HTTP status code: 401 (4xx = user error)
  - Helpful detail message: "API key is malformed, please type the API key correctly in the header."
  - Clear indication this is something they can fix (check API key)
  
  This shows concrete proof that the error handling works as intended.

File: src/ogx/providers/registry/file_processors.py (hunk 3 above)
Comment (mattf) [reply src/ogx/providers/registry/file_processors.py]: i think we can relax this - https://github.com/ogx-ai/ogx/pull/6131

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+        if not file and not file_id:
+            raise ValueError("Either file or file_id must be provided")
+        if file and file_id:
+            raise ValueError("Cannot provide both file and file_id")
+
+        start_time = time.time()
+
+        # Get file content and metadata
+        if file:
+            content = await file.read()
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py:61]: Might want to use the same `response_body_bytes` as docling-serve in case the raw bytes() cast breaks on certain response types. See: https://github.com/ogx-ai/ogx/blob/main/src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py#L74

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        log.info(
+            "Unstructured.io API returned elements",
+            filename=filename,
+            element_count=len(resp.elements),
+        )
+
+        # Convert response elements to list of dicts
+        return [dict(elem) for elem in resp.elements]
+
+    async def _partition_and_chunk(
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: `_partition_no_chunk` and `_partition_and_chunk` share a lot of code. Consider consolidating into a single private method with an optional chunking parameter to reduce duplication

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+        Returns:
+            List of element dictionaries from Unstructured API
+
+        Raises:
+            Exception: If API call fails
+        """
+        from unstructured_client import UnstructuredClient
+        from unstructured_client.models import operations, shared
+
+        client = UnstructuredClient(api_key_auth=self.config.api_key.get_secret_value())
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: Both _partion_no_chunk and _partition_and_chunk create a new UnstructuredClient. Consider creating the client once in __init__ or on first use.

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 8]
```diff
@@ -0,0 +1,283 @@
+# Copyright (c) The OGX Contributors.
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py:1]: Missing a shutdown method. See example: https://github.com/ogx-ai/ogx/blob/main/src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py#L225

File: tests/unit/providers/file_processor/test_unstructured_api.py [hunk 9]
```diff
@@ -0,0 +1,280 @@
+# Copyright (c) The OGX Contributors.
```
Comment (alinaryan) [inline tests/unit/providers/file_processor/test_unstructured_api.py:1]: All tests exercise the _partition_no_chunk path (no chunking_strategy in requests). The _partition_and_chunk method — including the auto/static strategy dispatch, max_tokens calculation, and by_title chunking— has no test coverage

File: docs/docs/providers/file_processors/remote_unstructured-api.mdx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+## Documentation
+
+See [Unstructured.io documentation](https://docs.unstructured.io) for API details and format support.
+
+
+## Configuration
+
+| Field | Type | Required | Default | Description |
+|-------|------|----------|---------|-------------|
+| `api_key` | `SecretStr` | No |  | API key for authenticating with Unstructured.io SaaS API (get one from https://unstructured.io) |
```
Comment (alinaryan) [inline docs/docs/providers/file_processors/remote_unstructured-api.mdx:136]: nit: consider adding a default

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+            )
+        )
+
+        log.info(
+            "Calling Unstructured.io API",
+            filename=filename,
+            size_bytes=len(content),
+        )
+
+        resp = client.general.partition(request=req)
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: This calls the synchronous partition() method inside an async def, which blocks the event loop for the duration of the API call. The unstructured-client SDK provides an async variant. swap to await client.general.partition_async(request=req) here and in _partition_and_chunk below.

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+
+        if not resp.elements:
+            log.warning("Unstructured.io API returned no elements", filename=filename)
+            return []
```
Comment (alinaryan) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: Consider adding tests to exercise the empty API response case

**Review by alinaryan — COMMENTED** [review]
Thanks for working on this! Left a few comments. Please also add tests to verify behavior when the Unstructured API returns errors (4xx, 5xx, network failures).

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 1 above)
Comment (mattf) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py:46]: @sahana-sreeram is correct

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 13]
```diff
@@ -0,0 +1,283 @@
...
+        document_metadata: dict[str, Any] = {"filename": filename}
+        if file_id:
+            document_metadata["file_id"] = file_id
+
+        # Call Unstructured.io API
+        if chunking_strategy:
+            log.info("Using chunking strategy", strategy_type=chunking_strategy.type)
+            elements = await self._partition_and_chunk(content, filename, chunking_strategy)
+        else:
+            log.info("No chunking strategy - using element-level chunks")
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: consider what an operator will see in the logs, consider moving these to debug

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+            filename: Original filename (used for format detection)
+
+        Returns:
+            List of element dictionaries from Unstructured API
+
+        Raises:
+            Exception: If API call fails
+        """
+        from unstructured_client import UnstructuredClient
+        from unstructured_client.models import operations, shared
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: imports at top please

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 7 above)
Comment (mattf) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: or at lease up a level.

we must not share api keys, so don't go all the way to `__init__`

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 12 above)
Comment (mattf) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: should this be an error case?

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 15]
```diff
@@ -0,0 +1,283 @@
...
+            partition_parameters=shared.PartitionParameters(
+                files=shared.Files(
+                    content=content,
+                    file_name=filename,
+                ),
+                strategy=shared.Strategy.AUTO,
+            )
+        )
+
+        log.info(
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: consider debug level

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+            size_bytes=len(content),
+        )
+
+        resp = client.general.partition(request=req)
+
+        if not resp.elements:
+            log.warning("Unstructured.io API returned no elements", filename=filename)
+            return []
+
+        log.info(
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: consider debug level

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+            chunking_strategy: Chunking configuration from request
+
+        Returns:
+            List of element dictionaries from Unstructured API (pre-chunked)
+
+        Raises:
+            Exception: If API call fails
+        """
+        from unstructured_client import UnstructuredClient
+        from unstructured_client.models import operations, shared
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: imports at top pls

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 18]
```diff
@@ -0,0 +1,283 @@
...
+                    content=content,
+                    file_name=filename,
+                ),
+                strategy=shared.Strategy.AUTO,
+                chunking_strategy="by_title",
+                max_characters=max_characters,
+            )
+        )
+
+        log.info(
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: consider debug level

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+            "Calling Unstructured.io API with chunking",
+            filename=filename,
+            size_bytes=len(content),
+            max_characters=max_characters,
+        )
+
+        resp = client.general.partition(request=req)
```
Comment (mattf) [inline src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: see @alinaryan's comment about async above

**Review by mattf — DISMISSED** [review]
`_partition_and_chunk` and `_partition_no_chunk` overlap. instead of separate funcs for making the request, do separate funcs for building the request.

```
def make_request(filename, content) -> PartitionRequest:
   ...

def make_request_with_chunking(filename, content, chunking_strategy) -> PartitionRequest:
   ...

if chunking_strategy:
   elements = await self._partition(self.make_request_with_chunking(filename, content, chunking_strategy))
else:
   elements = await self._partition(self.make_request(filename, content))
```

this gives you one place to manage the client construction, error handling, logging, async request handling, and post processing

**Comment by sahana-sreeram** [comment]
@alinaryan @mattf 

Changes: 
[d85d836](https://github.com/ogx-ai/ogx/pull/6076/commits/d85d83618f4ab633b392e518285ce5655a8a30a1) - Async handling and code style:
  - Use response_body_bytes for file_id path, mapping to docling_serve's implementation
  - Use partition_async instead of sync partition
  - Change log.info to log.debug throughout 
  - Add shutdown method
  
 [2bfa166](https://github.com/ogx-ai/ogx/pull/6076/commits/2bfa166c73a96efba302f65af6bf46d60f13fe8b) - Refactored client creation and consolidate partition methods:
  - Move client creation to process_file method.
  - Consolidate _partition_no_chunk and _partition_and_chunk -->
  - Separate request builders (_make_request, _make_request_with_chunking) from API execution (from Matt's comment above)

  [73755cc](https://github.com/ogx-ai/ogx/pull/6076/commits/73755cca476e4492c625e0ef19fb25f68f9e27b9) - Test coverage:
  - Add tests for chunking path coverage, auto/static strategies
  - Add tests verifying max token calculation, by_title chunking strategy
  - Add tests for empty API responses w and w/o chunking
  - Add error handling tests for 401/500 errors and network failures

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 7 above)
Comment (sahana-sreeram) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: [2bfa166](https://github.com/ogx-ai/ogx/pull/6076/commits/2bfa166c73a96efba302f65af6bf46d60f13fe8b)
Moved client creation to process_file method (moved up one level to consolidate, but not to init to avoid sharing API keys across requests per Matt's guidance). The client is created once at the start of process_file, then reused for the partition call

File: src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py (hunk 12 above)
Comment (sahana-sreeram) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: [73755cc](https://github.com/ogx-ai/ogx/pull/6076/commits/73755cca476e4492c625e0ef19fb25f68f9e27b9)
Kept this as a log (and added corresponding test) without being an error case to match [docling-serve](https://github.com/ogx-ai/ogx/blob/main/src/ogx/providers/remote/file_processor/docling_serve/docling_serve.py#L74) (lines 130-131, 187). I thought returning `[]` might make sense here because an empty response could be a valid blank/image-only document rather than an error.

Comment (mattf) [reply src/ogx/providers/remote/file_processor/unstructured_api/unstructured_api.py]: fair enough

**Comment by mattf** [comment]
@alinaryan let me know when you approve

**Review by alinaryan — APPROVED** [review]
Thank you for adding this!

**Review by mattf — APPROVED** [review]
per @alinaryan
